rock_paper_scissors/exc.py

```diff
@@ -0,0 +1,49 @@
+class ProgramError(Exception):
+    """Base exception class for errors raised by rock paper scissors program."""
+    pass
+
+
+class UserDefinedError(ProgramError):
+    """Raised for all user input errors."""
+    pass
+
+
+class UserChoiceInputError(UserDefinedError):
+    """Raised for invalid User sting input."""
+    def __init__(self, message):
+        self.message = message
+
+
+class UserRoundsInputError(UserDefinedError):
+    """Raised for invalid rounds input."""
+    pass
+
+
+class RoundsStringError(UserRoundsInputError):
+    """Raised if rounds input is a string."""
+    def __init__(self, message):
```

Comment:
It's actually discouraged to override `__init__` to implement a `self.message` attribute; `Exception.message` was [deprecated](http://legacy.python.org/dev/peps/pep-0352/) in Python 2.6.